File: algorithm/python/dp/Fibonacci.py
#-*-encoding:utf-8-*-
# fibonacci
# cww97
# 2017/09/26
from tracer import *
tracer = Array1DTracer('fib')
n = 15
fib = [0, 1]
for i in range(2, n):
    fib.append(0)
tracer.setData(fib)


if __name__ == '__main__':
    for i in range(2, n):
        # here is core code of fibonacci
        fib[i] = fib[i-1] + fib[i-2]
        tracer.select(i-2).select(i-1).wait()
        tracer.notify(i, fib[i]).wait()
        tracer.denotify(i).deselect(i-2).deselect(i-1)
    tracerlist.info()

File: algorithm/javascript/dp/Fibonacci.js
var tracer = new Array1DTracer('Sequence');
var index = 15;
var D = [1, 1];
for (var i = 2; i < index; i++) {
    D.push(0);
}
tracer._setData(D);


for (var i = 2; i < index; i++) {
    D[i] = D[i - 2] + D[i - 1];
    tracer._select(i - 2, i - 1)._wait();
    tracer._notify(i, D[i])._wait();
    tracer._denotify(i);
    tracer._deselect(i - 2, i - 1);
}